app/commands/authenticate_user.rb

```diff
@@ -1,31 +1,25 @@
 class AuthenticateUser
   prepend SimpleCommand
 
-  def initialize(email, password)
-    @email = email
-    @password = password
+  def initialize(user, valid_user_auth)
+    @user = user
+    @valid_user_auth = valid_user_auth
   end
 
   def call
-    if user
-      auth_token = JsonWebToken.encode(user_id: user.id) if user
-      return {
-        auth_token: auth_token, user: user.safe_attributes
-      }
-    else
-      raise 'Auth failed'
-    end
+    return invalid_credentials if @valid_user_auth.blank?
+
+    auth_token = JsonWebToken.encode(user_id: user.id)
+    { auth_token: auth_token, user: user.safe_attributes }
```

Comment:
I have gotten rid of `user.safe_atrributes` in my code that I haven't merged yet. Can you please change this to `as_json` to match the changes I will introduce later?
```suggestion
    { auth_token: auth_token, user: user.as_json }
```